packages/react-html/src/components/tests/HtmlUpdater.test.tsx

```diff
@@ -142,4 +142,38 @@ describe('<Provider />', () => {
     expect(metaElement).toBe(allMetas[0]);
     expect(allMetas[1].getAttribute('content')).toBe(metaTwo.content);
   });
+
+  it('only keeps the last added meta based on name or property', () => {
+    const globalDescription = 'global description';
+    const pageDescription = 'page description';
+
+    const latestDescription = {name: 'description', content: pageDescription};
+    const latestOgDescription = {
+      property: 'og:description',
+      content: pageDescription,
+    };
+
+    const manager = new HtmlManager();
+    mountWithManager(<HtmlUpdater />, manager);
+
+    manager.addMeta({name: 'description', content: globalDescription});
```

Comment:
Is there any reason why `HtmlManager` can't handle the dedupe instead of doing it afterwards?

Reply:
I tried that in my previous PR, see https://github.com/Shopify/quilt/pull/1732
`Html` and `HtmlUpdater` will both uses the `manager.state` so I originally thought it would be a nice idea to filter duplicates here but it does not work as expected.